datamodel/include/datamodel/math/MatrixColumnMajorAdaptor.hpp

```diff
@@ -1,25 +1,59 @@
 #ifndef RCBE_ENGINE_MATRIXCOLUMNMAJORADAPTOR_HPP
 #define RCBE_ENGINE_MATRIXCOLUMNMAJORADAPTOR_HPP
 
+#include <array>
 #include <cstdint>
+#include <iostream>
 
 #include <rcbe-engine/datamodel/math/Matrix.hpp>
 
+#ifdef RCBE_GLM_INTEGRATION
+#include <glm/matrix.hpp>
+#endif
+
+#define RCBE_DEBUG_MODE
+
 namespace rcbe::math {
+template <typename StoredValue = core::EngineScalar>
 class MatrixColumnMajorAdaptor {
 public:
-    using StorageType = std::vector<core::EngineScalar>;
+    using ValueType = StoredValue;
+    using StorageType = std::vector<ValueType>;
 
     MatrixColumnMajorAdaptor() = default;
     ~MatrixColumnMajorAdaptor() = default;
 
+#ifdef RCBE_GLM_INTEGRATION
+    template <glm::length_t r, glm::length_t c, typename T, glm::qualifier Q>
+    explicit MatrixColumnMajorAdaptor(const glm::mat<c, r, T, Q> &matrix)
+    {
+        adapted_storage_.reserve(r * c);
+        for (size_t i = 0; i < c; ++i) {
+            for (size_t j = 0; j < r; ++j) {
+                adapted_storage_.push_back(static_cast<ValueType>(matrix[i][j]));
+            }
+        }
+    }
+#endif
+
+    template <typename Value, size_t sz>
+    explicit MatrixColumnMajorAdaptor(std::array<Value, sz> storage)
+    {
+        for (const auto &s : storage) {
+            adapted_storage_.push_back(s);
+        }
+    }
+
+    explicit MatrixColumnMajorAdaptor(StorageType &&storage)
```

Comment:
This one is also strange can't tell what it does, so it basically takes column major storage and stores it inside this class?

Reply:
Added a comment